README.md

```diff
@@ -21,135 +21,169 @@ gem 'acme-client'
 ```
 
 ## Usage
+* [Setting up a client](#setting-up-a-client)
+* [Account management](#account-management)
+* [Obtaining a certificate](#obtaining-a-certificate)
+  * [Ordering a certificate](#ordering-a-certificate)
+  * [Completing an HTTP challenge](#preparing-for-http-challenge)
+  * [Completing an DNS challenge](#preparing-for-dns-challenge)
+  * [Request a challenge verification](#request-a-challenge-verification)
+  * [Downloading a certificate](#downloading-a-certificate)
+* [Extra](#extra)
+  * [Certificate revokation](#certificate-revokation)
+  * [Certificate renewal](#certificate-renewal)
 
-### Register client
+## Setting up a client
 
-In order to authenticate our client, we have to create an account for it.
+The client is initialized with a private key and the directory of your ACME provider.
+
+LetsEncrypt's `directory` is `https://acme-v01.api.letsencrypt.org/directory`
+
+`acme-ruby` expects `OpenSSL::PKey::RSA` or `OpenSSL::PKey::EC`
+
+You can generate one in Ruby using OpenSSL.
 
 ```ruby
-# We're going to need a private key.
 require 'openssl'
 private_key = OpenSSL::PKey::RSA.new(4096)
+```
 
-# We need an ACME server to talk to, see github.com/letsencrypt/boulder
-# WARNING: This endpoint is the production endpoint, which is rate limited and will produce valid certificates.
-# You should probably use the staging endpoint for all your experimentation:
-# endpoint = 'https://acme-staging.api.letsencrypt.org/'
-endpoint = 'https://acme-v01.api.letsencrypt.org/'
+Or load one from a PEM file
 
-# Initialize the client
-require 'acme-client'
-client = Acme::Client.new(private_key: private_key, endpoint: endpoint, connection_options: { request: { open_timeout: 5, timeout: 5 } })
+```ruby
+require 'openssl'
+OpenSSL::PKey::RSA.new(File.read('/path/to/private_key.pem'))
+```
+
+See [RSA](https://ruby.github.io/openssl/OpenSSL/PKey/RSA.html) and [EC](https://ruby.github.io/openssl/OpenSSL/PKey/EC.html) for documentation.
+
+
+```ruby
+client = Acme::Client.new(private_key: private_key, directory: 'https://acme-v01.api.letsencrypt.org/directory')
```

Comment:
Given this is targeting v2, the directory seems incorrect.

I would actually suggest to use the staging directory in the README as from my experience there is a risk users will just follow the readme blindly using production, without understanding there are strict limits there (and staging should be used for testing).

```
https://acme-staging-v02.api.letsencrypt.org/directory
```